pkg/helm/watcher/server.go

```diff
@@ -0,0 +1,94 @@
+package watcher
+
+import (
+	"io/ioutil"
+	"os"
+
+	sourcev1 "github.com/fluxcd/source-controller/api/v1beta1"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/healthz"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+
+	"github.com/weaveworks/weave-gitops/pkg/helm"
+	//+kubebuilder:scaffold:imports
+	"github.com/weaveworks/weave-gitops/pkg/helm/watcher/cache"
+	"github.com/weaveworks/weave-gitops/pkg/helm/watcher/controller"
+)
+
+var (
+	scheme   = runtime.NewScheme()
+	setupLog = ctrl.Log.WithName("setup")
+)
+
+type Watcher struct {
+	cache       cache.Cache
+	repoManager *helm.RepoManager
+}
+
+func NewWatcher(kubeClient client.Client, cache cache.Cache) (*Watcher, error) {
+	tempDir, err := ioutil.TempDir("", "helmrepocache")
+	if err != nil {
+		return nil, err
+	}
+
+	return &Watcher{
+		cache:       cache,
+		repoManager: helm.NewRepoManager(kubeClient, tempDir),
+	}, nil
+}
+
+func (w *Watcher) StartWatcher() error {
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+	utilruntime.Must(sourcev1.AddToScheme(scheme))
```

Comment:
worth moving to the `watcher` constructer, to prevent the code panicing after the go routine has started? seems a bit cleaner to fail earlier

Reply:
(I'd move it to the `main.go:main()` function which is what most other things do.)

Reply:
Good idea! 👍 Also, I'm actually gonna remove the `Must` call and return an error rather than panicking.